base/arithmetic.h

```diff
@@ -151,14 +152,45 @@ bool mul_overflows(T a, T b) {
   return __builtin_mul_overflow(a, b, &dummy);
 }
 
+/** Routines to perform arithmetic on signed types without triggering signed
+ * overflow. If overflow would occur, sets result to zero, and returns
+ * false. Otherwise set result to the correct value, and returns true. */
+template <class T>
+bool add_with_overflow(T a, T b, T* result) {
```

Comment:
I think either the name or the return value of these functions is negated. When it returns true, that means *without* overflow.

Maybe these should just be overloads of the above functions, except they actually pass the result instead of a dummy?

Also: I think the result should be a reference, not a pointer.

Reply:
This is how it was in Halide, so I just kept it. I flipped the value now and changed the result to a reference.